controllers/git_test.go

```diff
@@ -172,6 +173,54 @@ func TestPushRejected(t *testing.T) {
 	}
 }
 
+func TestEarlyEOF(t *testing.T) {
+	g := NewWithT(t)
+
+	gitServer, err := gittestserver.NewTempGitServer()
+	g.Expect(err).ToNot(HaveOccurred())
+	defer os.RemoveAll(gitServer.Root())
+
+	username := "norris"
+	password := "chuck"
+
+	gitServer.
+		AutoCreate().
+		KeyDir(filepath.Join(t.TempDir(), "keys")).
+		Auth(username, password).
+		ReadOnly(true)
+
+	err = gitServer.StartHTTP()
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = initGitRepo(gitServer, "testdata/appconfig", "test", "/appconfig.git")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	repoURL := gitServer.HTTPAddressWithCredentials() + "/appconfig.git"
+	cloneCtx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	access := repoAccess{
+		auth: &git.AuthOptions{
+			Username: username,
+			Password: password,
+		},
+	}
+
+	repo, err := clone(cloneCtx, repoURL, "test", access.auth)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	defer repo.Free()
+
+	cleanup, err := configureTransportOptsForRepo(repo, access.auth)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	defer cleanup()
+
+	err = push(context.TODO(), repo.Workdir(), "test", access)
+	g.Expect(err).To(HaveOccurred())
+	g.Expect(err.Error()).To(ContainSubstring("early EOF"))
```

Comment:
Since this test is supposed to test the enrichment of the "early EOF" error messages, it'd make more sense to check for the additional message that we explicitly add.